src/main/java/org/alfresco/repo/policy/EventBehaviour.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * #%L
+ * Alfresco Repository
+ * %%
+ * Copyright (C) 2005 - 2018 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software. 
+ * If the software was purchased under a paid Alfresco license, the terms of 
+ * the paid license agreement will prevail.  Otherwise, the software is 
+ * provided under the following open source license terms:
+ * 
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.repo.policy;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+
+import org.alfresco.api.AlfrescoPublicApi;
+import org.alfresco.repo.rawevents.EventProducer;
+import org.springframework.extensions.surf.util.ParameterCheck;
+
+/**
+ * Event based Behaviour.
+ * 
+ * The event behavior delegates the generation of the event to a method pointer.  The pointer is
+ * represented by an instance object and method name.
+ *
+ * @author Cristian Turlica
+ */
+@AlfrescoPublicApi
+public class EventBehaviour extends BaseBehaviour
+{
+
+    EventProducer eventProducer;
```

Comment:
Are these variables intentionally package-private?

Reply:
Changed to private for now.